### PR TITLE
Web apps

### PR DESCRIPTION
**Pull request by fdistorted**
Some relations are 1 to 1 as they are big I have decided to place them in separate tables

### CODE REVIEW

File: client/services/services.go [hunk 1]
```diff
@@ -3,10 +3,12 @@
 //go:generate mockgen -destination=./mocks/ad_groups.go -package=mocks . ADGroupsClient
 //go:generate mockgen -destination=./mocks/ad_service_principals.go -package=mocks . ADServicePrinicpals
 //go:generate mockgen -destination=./mocks/ad_users.go -package=mocks . ADUsersClient
+//go:generate mockgen -destination=./mocks/network.go -package=mocks . WatchersClient,VirtualNetworksClient
 //go:generate mockgen -destination=./mocks/monitor.go -package=mocks . LogProfilesClient
-//go:generate mockgen -destination=./mocks/network.go -package=mocks . VirtualNetworksClient,SecurityGroupsClient,WatchersClient
```
Comment (irmatov) [inline client/services/services.go:7]: SecurityGroupsClient dropped unintentionally?

Comment (fdistorted) [reply client/services/services.go:7]: might be. thanks

**Review by irmatov — APPROVED** [review]
LGTM

File: resources/web_app_auth_settings.go [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+			},
+			{
+				Name:        "app_id",
+				Description: "Unique ID of azure_web_apps table (FK)",
+				Type:        schema.TypeString,
+				Resolver:    schema.ParentResourceFieldResolver("id"),
+			},
+			{
+				Name:        "enabled",
+				Description: "otherwise, <code>false</code>",
```
Comment (roneli) [inline resources/web_app_auth_settings.go:33]: bad description

File: resources/web_app_auth_settings.go [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+func WebAppAuthSettings() *schema.Table {
+	return &schema.Table{
+		Name:        "aws_web_app_auth_settings",
+		Description: "SiteAuthSettings configuration settings for the Azure App Service Authentication / Authorization feature",
+		Resolver:    fetchWebAppAuthSettings,
+		Options:     schema.TableCreationOptions{PrimaryKeys: []string{"app_cq_id", "id"}},
+		Columns: []schema.Column{
+			{
+				Name:        "app_cq_id",
+				Description: "Unique ID of azure_web_apps table (FK)",
```
Comment (roneli) [inline resources/web_app_auth_settings.go:21]: Unique CloudQuery ID

File: resources/web_apps.go [hunk 4]
```diff
@@ -0,0 +1,1675 @@
...
+					},
+					{
+						Name:        "description",
+						Description: "IP restriction rule description",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "headers",
+						Description: "IP restriction rule headers X-Forwarded-Host (https://developermozillaorg/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host#Examples) The matching logic is  - If the property is null or empty (default), all hosts(or lack of) are allowed - A value is compared using ordinal-ignore-case (excluding port number) - Subdomain wildcards are permitted but don't match the root domain For example, *contosocom matches the subdomain foocontosocom  but not the root domain contosocom or multi-level foobarcontosocom - Unicode host names are allowed but are converted to Punycode for matching X-Forwarded-For (https://developermozillaorg/en-US/docs/Web/HTTP/Headers/X-Forwarded-For#Examples) The matching logic is  - If the property is null or empty (default), any forwarded-for chains (or lack of) are allowed - If any address (excluding port number) in the chain (comma separated) matches the CIDR defined by the property X-Azure-FDID and X-FD-HealthProbe The matching logic is exact match",
```
Comment (roneli) [inline resources/web_apps.go:1427]: shorten description

File: resources/web_apps.go [hunk 5]
```diff
@@ -0,0 +1,1675 @@
...
+					{
+						Name:        "ip_address",
+						Description: "IP address the security restriction is valid for It can be in form of pure ipv4 address (required SubnetMask property) or CIDR notation such as ipv4/mask (leading bit match) For CIDR, SubnetMask property must not be specified",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("IPAddress"),
+					},
+					{
+						Name:        "subnet_mask",
+						Description: "Subnet mask for the range of IP addresses the restriction is valid for",
+						Type:        schema.TypeString,
```
Comment (roneli) [inline resources/web_apps.go:1382]: You can try the new IP/INET etc' types

File: resources/web_apps.go [hunk 6]
```diff
@@ -0,0 +1,1675 @@
...
+					},
+					{
+						Name:        "description",
+						Description: "IP restriction rule description",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "headers",
+						Description: "IP restriction rule headers X-Forwarded-Host (https://developermozillaorg/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host#Examples) The matching logic is  - If the property is null or empty (default), all hosts(or lack of) are allowed - A value is compared using ordinal-ignore-case (excluding port number) - Subdomain wildcards are permitted but don't match the root domain For example, *contosocom matches the subdomain foocontosocom  but not the root domain contosocom or multi-level foobarcontosocom - Unicode host names are allowed but are converted to Punycode for matching X-Forwarded-For (https://developermozillaorg/en-US/docs/Web/HTTP/Headers/X-Forwarded-For#Examples) The matching logic is  - If the property is null or empty (default), any forwarded-for chains (or lack of) are allowed - If any address (excluding port number) in the chain (comma separated) matches the CIDR defined by the property X-Azure-FDID and X-FD-HealthProbe The matching logic is exact match",
```
Comment (roneli) [inline resources/web_apps.go:1350]: shorten description

File: resources/web_apps.go [hunk 7]
```diff
@@ -0,0 +1,1675 @@
...
+			},
+			{
+				Name:        "azure_web_app_ip_security_restrictions",
+				Description: "IPSecurityRestriction IP security restriction on an app",
+				Resolver:    fetchWebAppIpSecurityRestrictions,
+				Options:     schema.TableCreationOptions{PrimaryKeys: []string{"app_cq_id", "name"}},
+				Columns: []schema.Column{
+					{
+						Name:        "app_cq_id",
+						Description: "Unique ID of azure_web_apps table (FK)",
```
Comment (roneli) [inline resources/web_apps.go:1286]: Unique CloudQuery ID

File: resources/web_apps.go [hunk 8]
```diff
@@ -0,0 +1,1675 @@
...
+)
+
+func WebApps() *schema.Table {
+	return &schema.Table{
+		Name:        "azure_web_apps",
+		Description: "Site a web app, a mobile app backend, or an API app",
+		Resolver:    fetchWebApps,
+		Multiplex:   client.SubscriptionMultiplex,
+		Options:     schema.TableCreationOptions{PrimaryKeys: []string{"subscription_id", "id"}},
+		Columns: []schema.Column{
```
Comment (roneli) [inline resources/web_apps.go:21]: I think we can reduce relations/sub-relations into json types.

**Review by roneli — CHANGES_REQUESTED** [review]
Would reduce the amount of unimportant relations (relations that don't hold important data) into json fields

File: resources/web_app_auth_settings.go (hunk 3 above)
Comment (roneli) [reply resources/web_app_auth_settings.go:21]: update description as well

File: resources/web_apps.go [hunk 9]
```diff
@@ -0,0 +1,558 @@
...
+					{
+						Name: "publish_url",
+						Type: schema.TypeString,
+					},
+					{
+						Name: "user_name",
+						Type: schema.TypeString,
+					},
+					{
+						Name: "user_p_w_d",
```
Comment (roneli) [inline resources/web_apps.go:465]: user_pwd

File: resources/web_app_auth_settings.go [hunk 10]
```diff
@@ -0,0 +1,300 @@
...
+				Type:        schema.TypeUUID,
+				Resolver:    schema.ParentIdResolver,
+			},
+			{
+				Name:        "app_id",
+				Description: "Unique CloudQuery ID of azure_web_apps table (FK)",
```
Comment (roneli) [inline resources/web_app_auth_settings.go:27]: the other way round this is the app_id is the original resource id and the app_cq_id is the unique cloudquery id

**Review by roneli — CHANGES_REQUESTED** [review]
Some column changes are required, but looks great with relation tables reduction! good job!

File: resources/web_app_auth_settings.go (hunk 3 above)
Comment (roneli) [inline resources/web_app_auth_settings.go:21]: Fix this description please :)

**Review by roneli — APPROVED** [review]
Looks good, fix minor description and we can merge this.

**Pull request merged**